### PR TITLE
Feature/feature achievements

### PR DESCRIPTION
**Pull request by AlbertPG**

### CODE REVIEW

File: app/src/test/java/org/pesmypetcare/mypetcare/services/StubAchievementService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * @author Daniel Clemente
+ */
+public class StubAchievementService implements AchievementService {
+    private Map<String, List<UserMedalData>> medals;
+
+    public StubAchievementService() {
+        medals = new TreeMap<>();
+        UserMedalData userMedalData = new UserMedalData("Medal", Arrays.asList(10.0, 20.0, 30.0), "Test", new byte[0],
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/services/StubAchievementService.java:22]: Codacy Issue found: ['10.0' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5653081)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/achievements/TestTrUpdateMedalProgress.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/achievements/TestTrGetAllAchievements.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/achievements/AchievementsControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/achievements/TrUpdateMedalProgress.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubAchievementService.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/achievements/TrGetAllAchievements.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/features/users/UserAchievement.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/achievements/AchievementsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/services/achievement/AchievementAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/achievements/AchievementsComponentView.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5653081&bid=18509390)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/achievements/AchievementsComponentView.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/achievements/TestTrGetAllAchievements.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubAchievementService.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/achievements/AchievementsControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/users/UserAchievement.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/achievements/TestTrUpdateMedalProgress.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/achievement/AchievementAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/achievements/TrGetAllAchievements.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/achievements/AchievementsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/achievements/TrUpdateMedalProgress.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5653081&bid=18509390)

**Review by santidrj — DISMISSED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/features/users/UserAchievement.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/achievements/TestTrGetAllAchievements.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/achievements/TestTrUpdateMedalProgress.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/achievements/TrGetAllAchievements.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/achievements/AchievementsControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/achievements/AchievementsView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/views/circularentry/achievements/AchievementsComponentView.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/services/achievement/AchievementAdapter.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubAchievementService.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/achievements/TrUpdateMedalProgress.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5653081&bid=18509390)

**Review by santidrj — APPROVED** [review]
:+1:

**Review by marclll — APPROVED** [review]

**Pull request merged**